src/test/java/com/taspro/base/DriverFactory.java

```diff
@@ -0,0 +1,53 @@
+package com.taspro.base;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+
+import com.taspro.utility.PropertiesUtil;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+
+public class DriverFactory {
+
+	private static final PropertiesUtil prop = new PropertiesUtil();
+	private static final DriverFactory instance = new DriverFactory();
+
+	private static ThreadLocal<WebDriver> tl = new ThreadLocal<WebDriver>();
+
+	private DriverFactory() {
+
+	}
+
+	public static DriverFactory getDriverFactoryInstance() {
+		return instance;
+	}
+
+	public WebDriver initializeDriver() {
```

Comment:
add string parameter to get the browser name.

Reply:
Done.